src/app/query/shared/query-and-result-sections/query-and-result-sections.component.ts

```diff
@@ -130,6 +142,15 @@ export class QueryAndResultSectionsComponent extends BaseComponent implements On
         if (initialSqlRun) {
             this.resetRowsLimits();
         }
+        if (this.enabledProof) {
+            this.queryExplainerService
+                .processQueryWithProof(this.sqlRequestCode)
```

Comment:
My understanding is that when `Generate proof` toggle is on we will:
- Execute REST `/query` call that returns query result + proof
- Then we execute GraphQL `{ data { query() { ... } }` call which will run the SQL query AGAIN

i.e. we're using REST API to get proof and GQL to get data.

This is not only wasteful, but may result in a proof that doesn't actually correspond to data if the SQL query is non-deterministic (e.g. has random row order).

We should either switch to REST API for all queries, or extend GQL API to support proofs too.

Reply:
Will leave the decision to you and @zaychenko-sergei.

From my perspective it's also Ok to merge this feature in its current (bipolar) state, and do a proper fix later.

Reply:
I don't see urgency in this correction, so I prefer not to merge it in the intermediate state. With the API extension, the implementation of data access code will change signifficantly.

Reply:
There is some urgency to this. This PR makes query verifiability easier to demonstrate to potential users, so I'd like to get this in ideally in the next 1-2 weeks.

Let me think about proofs and GQL API more. Proofs rely on precise formatting of JSON (canonical) so I'm not 100% sure REST proofs can be adopted easily to GQL. They are unfortunately protocol-dependant.